spec/ScorecardSpec.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+describe('Scorecard', function() {
+  var scorecard;
+
+  beforeEach(function() {
+    scorecard = new Scorecard();
+  });
+
+  it('starts with empty scorecard', function() {
+    expect(scorecard.getFrames()).toEqual([]);
+  });
+
+  it('hold rolls in current frame until frame complete', function() {
+    scorecard.roll(3)
+
+    expect(scorecard.getCurrentFrame()).toEqual([3])
+  });
+
+  it('saves rolls as frames and stores together', function() {
+    scorecard.roll(3)
+    scorecard.roll(4)
+    scorecard.roll(2)
+    scorecard.roll(5)
+
+    expect(scorecard.getFrames()).toEqual([[3, 4], [2, 5]])
+  });
+
+  it('adds points to total once frame is complete', function() {
+    scorecard.roll(3)
+    scorecard.roll(4)
+    scorecard.roll(2)
+    scorecard.roll(5)
+
+    expect(scorecard.getTotal()).toEqual(14)
+  });
+
+  it('saves strike in frame alone', function() {
+    scorecard.roll(10)
+
+    expect(scorecard.getFrames()).toEqual([[10]])
+  })
+
+  it('gives bonus for one strike - next two rolls doubled', function() {
+    scorecard.roll(10)
+    scorecard.roll(2)
+    scorecard.roll(6)
+
+    expect(scorecard.getTotal()).toEqual(26)
+  });
+
+  it('gives bonus for spare - next roll doubled', function() {
+    scorecard.roll(5)
+    scorecard.roll(5)
+    scorecard.roll(2)
+    scorecard.roll(6)
+
+    expect(scorecard.getTotal()).toEqual(20)
+  })
+
+  it('gives bonus for double strike - two rolls after strikes doubles', function() {
+    scorecard.roll(10)
+    scorecard.roll(10)
+    scorecard.roll(2)
+    scorecard.roll(6)
+
+    expect(scorecard.getTotal()).toEqual(48)
+  })
+
+  it('works with ten frames', function() {
+    scorecard.roll(1)
```

Comment:
You can check which ones are repetitive and put it in a (for) loop, I can see a lot of `4`s.